### PR TITLE
[CPU] [TRANSFORMATIONS] FakeConvert decomposition transformation

### PR DESCRIPTION
**Pull request by xuchen-intel**
### Details:
 - *Implement FakeConvert decomposition transformation.*

### Tickets:
 - *[CVS-156963](https://jira.devtools.intel.com/browse/CVS-156963)*

### Prerequisites:
- *https://github.com/openvinotoolkit/openvino/pull/27949*

### CODE REVIEW

File: src/common/transformations/src/transformations/op_conversions/fake_convert_decomposition.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        if (fake_convert_node == nullptr || transformation_callback(fake_convert_node)) {
+            return false;
+        }
+
+        Output<Node> data{fake_convert_node->input_value(0)};
+        const Output<Node> input_scale{fake_convert_node->input_value(1)};
+        auto input_type = data.get_element_type();
+
+        ov::NodeVector decomp_ops;
```
Comment (praasz) [inline src/common/transformations/src/transformations/op_conversions/fake_convert_decomposition.cpp:37]:
```suggestion
        ov::pass::NodeRegistry decomp_ops;
```
use it instead then :

```cpp
  if (input_type != input_scale.get_element_type()) {
            input_type = input_scale.get_element_type();
            decomp_ops.make<ov::op::v0::Convert>(data, input_type); // it returns created node which can be used later if required
        }
/// at end
  ov::copy_runtime_info(fake_convert_node, decomp_ops.get());
```

File: src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+protected:
+    void SetUp() override {
+        FakeConvertDecompositionParams params = this->GetParam();
+
+        Shape data_shape, scale_shape, shift_shape;
+        element::Type_t data_prec, dst_prec;
+        bool default_shift;
+        std::tie(data_shape, scale_shape, shift_shape, data_prec, dst_prec, default_shift) = params;
+
+        std::shared_ptr<ov::Model> f(nullptr);
```
Comment (praasz) [inline src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp:56]:
```suggestion
        std::shared_ptr<ov::Model> model;
```
Detail: use model instead `f`

File: src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                                          ? std::make_shared<opset13::FakeConvert>(data, scale, dst_prec)
+                                          : std::make_shared<opset13::FakeConvert>(data, scale, shift, dst_prec);
+            f = std::make_shared<ov::Model>(NodeVector{fake_convert}, ParameterVector{data});
+
+            pass::Manager manager;
+            manager.register_pass<ov::pass::InitNodeInfo>();
+            manager.register_pass<ov::pass::FakeConvertDecomposition>();
+            manager.run_passes(f);
+
+            OV_ASSERT_NO_THROW(check_rt_info(f));
```
Comment (praasz) [inline src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp:72]: Can test be added in test body instead test's setup?

File: src/common/transformations/include/transformations/op_conversions/fake_convert_decomposition.hpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * f8: f8e4m3, f8e5m2
+ * downconvert: f32->f8, f16->f8, bf16->f8
+ * upconvert: f8->f32, f8->f16, f8->bf16
+ * output = (upconvert(downconvert(input * scale - shift)) + shift) / scale
+ *
+ */
+
+class ov::pass::FakeConvertDecomposition : public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("FakeConvertDecomposition", "0");
```
[the remark this answers is not in the document]
Comment (xuchen-intel) [reply src/common/transformations/include/transformations/op_conversions/fake_convert_decomposition.hpp]: Applied. Thanks Tomasz!

File: src/common/transformations/src/transformations/op_conversions/fake_convert_decomposition.cpp (hunk 1 above)
Comment (xuchen-intel) [reply src/common/transformations/src/transformations/op_conversions/fake_convert_decomposition.cpp]: Applied. Thanks Pawel!

File: src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp (hunk 2 above)
Comment (xuchen-intel) [reply src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp]: Applied. Thanks Pawel!

File: src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp (hunk 3 above)
Comment (xuchen-intel) [reply src/common/transformations/tests/op_conversions/fake_convert_decomposition_test.cpp]: Applied. Thanks Pawel!

**Comment by xuchen-intel** [comment]
@praasz @t-jankowski Hi Pawel and Tomasz, do you have more comments?

**Comment by xuchen-intel** [comment]
@dmitry-gorokhov Hi Dmitry, could you please also take a look?

File: src/common/transformations/include/transformations/op_conversions/fake_convert_decomposition.hpp [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+}  // namespace pass
+}  // namespace ov
+
+/**
+ * @ingroup ov_transformation_common_api
+ * @brief FakeConvertDecomposition transformation decomposes FakeConvert layer.
+ * f8: f8e4m3, f8e5m2
+ * downconvert: f32->f8, f16->f8, bf16->f8
+ * upconvert: f8->f32, f8->f16, f8->bf16
```
Comment (dmitry-gorokhov) [inline src/common/transformations/include/transformations/op_conversions/fake_convert_decomposition.hpp:23]: Only these precision are supported by decomposition transformation or this is just an example?

**Comment by dmitry-gorokhov** [comment]
@xuchen-intel Could you please also implement shared single layer test for FakeConvert operation and add its instance for CPU plugin?

File: src/common/transformations/include/transformations/op_conversions/fake_convert_decomposition.hpp (hunk 5 above)
Comment (xuchen-intel) [reply src/common/transformations/include/transformations/op_conversions/fake_convert_decomposition.hpp:23]: Thanks Dmitry for the comment! Yes, they are the only supported precisions, which are assured by https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/op/fake_convert.cpp#L90-L110 and https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/op/fake_convert.cpp#L129-L134.

**Comment by xuchen-intel** [comment]
> @xuchen-intel Could you please also implement shared single layer test for FakeConvert operation and add its instance for CPU plugin?

Applied. Thanks Dmitry!

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**